File: java/Figuras/src/main/java/org/example/Figura3d.java
package org.example;

/**
 * Clase abstracta para figuras tridimensionales
 */
public abstract class Figura3d extends Figura {
    
    /**
     * Constructor de la clase Figura3d
     * @param nombre Nombre de la figura
     * @param idFigura Identificador único de la figura
     */
    public Figura3d(String nombre, int idFigura) {
        super(nombre, "3D", idFigura);
    }
    
    /**
     * Calcula el volumen de la figura 3D
     * @return El volumen de la figura
     */
    public abstract double calcularVolumen();
}

File: python/Figuras/Figura3d.py
"""
Clase abstracta para figuras tridimensionales
"""
# pylint: disable=invalid-name
from abc import abstractmethod
from Figura import Figura

class Figura3d(Figura):
    """Clase abstracta para figuras tridimensionales"""

    def __init__(self, nombre: str, id_figura: int):
        """
        Constructor de la clase Figura3d

        Args:
            nombre (str): Nombre de la figura
            id_figura (int): Identificador único de la figura
        """
        super().__init__(nombre, "3D", id_figura)

    @abstractmethod
    def calcular_volumen(self) -> float:
        """Calcula el volumen de la figura 3D"""
        pass
